### ISSUE TITLE
video torrents

### ISSUE THREAD

**Issue by markvdb** [FOSDEM/video#79]
We probably want torrents for the videos.

**Comment by yoe** [comment]
Would be nice.

I've also been thinking about adding a structured format (e.g., JSON) to video.f.o that describes all (or most) of the videos available (with at least their title and subtitle, possibly also a description etc). This would allow things like a kodi plugin etc.

**Comment by johanvdw** [comment]
We already have the schedule xml - imho makes more sense to add extra video information there.

**Comment by yoe** [comment]
That would work, but requires more work for clients to get all the data. On top of that, would make it harder for us to update metadata for past years where it currently doesn't exist.

I think adding a parsable file on video.f.o is easier. It is also what most other conferences do.

**Comment by krokodilerian** [comment]
Comment from @markvdb  in #83 
"
Plus metadata. Should be watchable through this, in web based viewers, popcorn time style. Should be a nice example of legal torrents use. A nice FOSS "services" example too, a frontier area really of FOSS software.

The nec plus ultra as far as I'm concerned should be a way to watch FOSDEM videos online without real central server infrastructure except for one of us serving the torrents from a home file server as the last resort...
"

**Comment by SWATx** [comment]
I'll have a look into creating these and will let you know about the progress.

**Comment by markvdb** [comment]
Peertube would be fun too, but all of this is wishlist really.

**Comment by markvdb** [comment]
Another option that @RichiH mentioned was to look for suitable instances that could host FOSDEM video.

I compared a pile of peertube instances. https://framatube.org/about/instance looks like the most suitable one. It's hosted by framasoft, the peertube developers. Framasoft is a non-profit focused on free software. They've been around since 2001 and seem to be as energetic as ever.

**Comment by 22decembre** [comment]
Hello. I am seriously interested in having Fosdem videos on PeerTube. I think Framasoft would prefer not to host your videos, because nowaydays, everybody rely on them (for their services) instead of mounting their own fediverse instances, which is ultimately Framasoft's goal.

Why don't you mount your own instance ?

I have my own ( https://tube.22decembre.eu ). I can host some of your videos. But I feel you would be better served yourself. Then we could all subscribe to your instance.

Thank you so much.

**Comment by yoe** [comment]
The reason we don't do this is, mostly, because we don't have the infrastructure to support that. We already store our videos on a number of mirrors, and video.fosdem.org already contains several hundreds of gigabytes of recordings. As it is, storing that would require us to duplicate the same video into a peertube instance, for which we just don't have the storage.

If it would be possible to run some server that exports metadata of our videos into the fediverse *without* re-uploading everything, we'd be all over that. Unfortunately, that doesn't exist today.

**Comment by rzr** [comment]
soon or later some of them will be mirrored at various places

If they are well tagged (#FOSDEM2021) they should appear on

https://sepiasearch.org/search?search=fosdem2021&sort=-match&page=1

I also suggested to conf.tube to mirror them:

https://mastodon.social/@rzr/105695289912075310

I've seen also a couple (with Q&A) at:

https://www.youtube.com/hashtag/fosdem2021

**Comment by saerdnaer** [comment]
> If it would be possible to run some server that exports metadata of our videos into the fediverse without re-uploading everything, we'd be all over that. Unfortunately, that doesn't exist today.

Yes, definitly a feature that Peertube should implement – we had the same issue with in context of media.ccc.de

**Comment by markvdb** [comment]
from the matrix channel: "So maybe there is a simpler solution, then hosting a full peertube instance, if someone could "just" implement an activitypub actor that can be followed on the mirror server.
https://blog.joinmastodon.org/2018/06/how-to-implement-a-basic-activitypub-server/ < something.. "

**Comment by yoe** [comment]
On Sun, Feb 26, 2023 at 12:13:30AM -0800, Mark Van den Borre wrote:
>    from the matrix channel: "So maybe there is a simpler solution, then
>    hosting a full peertube instance, if someone could "just" implement an
>    activitypub actor that can be followed on the mirror server.
>    [1]https://blog.joinmastodon.org/2018/06/how-to-implement-a-basic-activ
>    itypub-server/ < something.. "

Yes, this is exactly what I have been suggesting to you for <checks
notes> three years now... but you always said it was a bad idea ;-)

I have looked at that blog post (and various other bits of
documentation) a few times, and have started looking at the available
bits of support code (e.g., Mojolicious::Plugin::WebFinger is one) that
would help with implementing the necessary items.

It's going to be quite the effort, but hey.

-- 
     ***@***.***{be,co.za}
wouter@{grep.be,fosdem.org,debian.org}

I will have a Tin-Actinium-Potassium mixture, thanks.

**Comment by yoe** [comment]
i.e., this:

> If it would be possible to run some server that exports metadata of our videos into the fediverse _without_ re-uploading everything, we'd be all over that. Unfortunately, that doesn't exist today.

which I said in 2020 in this very issue is exactly the same thing :-)

**Comment by testman42** [comment]
I assume the issue persists after all these years?

**Comment by Bugsbane** [comment]
So the issue is not wanting to duplicate video files, right? So, rather than waiting years for new server software, how about just leaving previous years videos as they are putting the videos from 2025 onwards in a Peertube instance?

No duplication or extra storage of video files.
2025+ videos are available on The Fedi.
No need to wait for new software to be developed.

**Comment by JLP** [comment]
+1 I would also like to watch the videos from PeerTube and comment on them from my Mastodon account

**Comment by schmaker** [comment]
Good news everyone: [FOSDEM 2026 on PeerTube](https://schmaker.eu/display/c83e3896-2169-6d53-c71f-64c343942764)

There will be experimental mesh from PeerTube instances streaming few FOSDEM rooms on PeerTube as well :)

**Comment by markvdb** [comment]
https://github.com/FOSDEM/video/commit/679c88a805cd5c37d60adf3614de06d499ff3722 is a first experiment towards a minimal ActivityPub + PeerTube extensions server. Helping hands welcome to troubleshoot the problems I bumped into.

Suspicion is on the webm present in the video links. Thanks to a suggestion by B D in #peertube:matrix.org .
